ast/src/analyzed/mod.rs

```diff
@@ -631,11 +633,18 @@ impl PublicDeclaration {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize, JsonSchema)]
-pub struct Identity<Expr> {
+#[derive(
+    Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy, Serialize, Deserialize, JsonSchema,
+)]
+pub struct IdentityId {
     /// The ID is specific to the identity kind.
-    pub id: u64,
+    pub local_id: u64,
     pub kind: IdentityKind,
```

Comment:
I don't really like moving the kind into the id structure.

I mean in the end, the ID only gets relevant once you group identities. If we later on construct identities as PIL structs, they won't receive an ID right away because we can still modify them. Only in or after the condenser will they get an ID. Maybe we should pull the id out of the struct instead of moving integral parts of the struct into the id?

Reply:
I actually just realized that the IDs currently aren't even unique. The condenser does not re-assign them.

Reply:
But isn't that analogous to how we currently have:
```rust
pub struct PolyID {
    pub id: u64,
    pub ptype: PolynomialType,
}
```

I think it makes sense to have some kind of unique ID for all identities. Should we fix this in a different PR and then base this PR onto that?